ffi/dart/lib/src/hook_helpers/build_libs.dart

```diff
@@ -85,26 +85,28 @@ Future buildLib(
   final additionalFeatures = isNoStd
       ? ['libc_alloc', 'looping_panic_handler']
       : ['simple_logger'];
-  await runProcess('cargo', [
-    if (buildStatic || isNoStd) '+$nightly',
-    'rustc',
-    '--manifest-path=ffi/capi/Cargo.toml',
-    '--crate-type=${buildStatic ? 'staticlib' : 'cdylib'}',
-    '--release',
-    '--config=profile.release.panic="abort"',
-    '--config=profile.release.codegen-units=1',
-    '--no-default-features',
-    '--features=${{...cargoFeatures, ...additionalFeatures}.join(',')}',
-    if (isNoStd) '-Zbuild-std=core,alloc',
-    if (buildStatic || isNoStd) ...[
-      '-Zbuild-std=std,panic_abort',
-      '-Zbuild-std-features=panic_immediate_abort',
+  await runProcess(
+    'cargo',
+    [
+      if (buildStatic || isNoStd) '+$nightly',
+      'rustc',
+      '--manifest-path=ffi/capi/Cargo.toml',
+      '--crate-type=${buildStatic ? 'staticlib' : 'cdylib'}',
+      '--release',
+      '--config=profile.release.panic="abort"',
+      '--config=profile.release.codegen-units=1',
+      '--no-default-features',
+      '--features=${{...cargoFeatures, ...additionalFeatures}.join(',')}',
+      if (isNoStd) '-Zbuild-std=core,alloc',
+      if (buildStatic || isNoStd) ...['-Zbuild-std=std,panic_abort'],
+      '--target=$rustTarget',
+      '--',
+      '--emit',
+      'link=${out.toFilePath(windows: Platform.isWindows)}',
     ],
-    '--target=$rustTarget',
-    '--',
-    '--emit',
-    'link=${out.toFilePath(windows: Platform.isWindows)}',
-  ], workingDirectory: workingDirectory);
+    workingDirectory: workingDirectory,
+    environment: {'RUSTFLAGS': '-Zunstable-options -Cpanic=immediate-abort'},
```

Comment:
Observation: these RUSTFLAGS are not conditional on anything, unlike the command above which makes them conditional on `buildStatic` or `isNoStd`